pex/common.py

```diff
@@ -410,7 +418,18 @@ def unlock():
         # N.B.: Since lockf operates on an open file descriptor and these are guaranteed to be
         # closed by the operating system when the owning process exits, this lock is immune to
         # staleness.
-        fcntl.lockf(lock_fd, fcntl.LOCK_EX)  # A blocking write lock.
+        if WINDOWS:
+            while True:
+                # Force the non-blocking lock to be blocking. LK_LOCK is msvcrt's implementation of
```

Comment:
Presumably then using `LK_LOCK` is a bit better here. You still need to loop, but presumably its slightly more efficient to let Windows do the 1st 10 tries before sleeping here and looping than to do the 1st 10 tries here manually.

Reply:
Makes sense, it also raises a clearer error (`OSError: [Errno 36] Resource deadlock avoided`) when the locking operation times out after 10 tries, so we can use that directly rather than an `OSError` catchall. Fixed in https://github.com/pantsbuild/pex/pull/1157/commits/edc84354c38a4d422fc551f58564a8323cadbf06

Reply:
Excellent - that's much better.